packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -1428,13 +1428,15 @@ class RenderEditable extends RenderBox {
       final TextSelection lastWord = to == null ?
         firstWord : _selectWordAtOffset(_textPainter.getPositionForOffset(globalToLocal(to - _paintOffset)));
 
-      onSelectionChanged(
-        TextSelection(
-          baseOffset: firstWord.base.offset,
-          extentOffset: lastWord.extent.offset,
-          affinity: firstWord.affinity,
-        ), this, cause,
+      final TextSelection nextSelection = TextSelection(
+        baseOffset: firstWord.base.offset,
+        extentOffset: lastWord.extent.offset,
+        affinity: firstWord.affinity,
       );
+      if (nextSelection == selection) {
+        return;
+      }
```

Comment:
@HansMuller Why isn't onSelectionChanged doing this sort of check already?  Should this be there instead?

Reply:
It could be. However since we know what the current selection was, I would think that we shouldn't be calling onSelection**Changed** unless the selection actually changed.

There are loads of calls to onSelectionChanged here. Maybe they should all be replaced with `_maybeCallOnSelectionChanged`

Reply:
Good idea, I'll try it.